File: my_vpn/app/src/main/java/com/myvpn/simple/xray/XrayConfig.java
package com.myvpn.simple.xray;

import java.util.List;
import java.util.Map;

public class XrayConfig {
    public List<Inbound> inbounds;
    public List<Outbound> outbounds;
    public Log log = new Log();
    public Routing routing;
    
    public static class Log {
        public String loglevel = "warning";
    }
    
    public static class Inbound {
        public String protocol;
        public int port;
        public String listen;
        public Map<String, Object> settings;
        public Sniffing sniffing;
    }
    
    public static class Outbound {
        public String tag;
        public String protocol;
        public Object settings;
        public StreamSettings streamSettings;
    }
    
    public static class Sniffing {
        public boolean enabled;
        public List<String> destOverride;
        public boolean routeOnly;
    }
    
    public static class StreamSettings {
        public String network;
        public String security;
        public TlsSettings tlsSettings;
    }
    
    public static class TlsSettings {
        public String serverName;
        public boolean allowInsecure;
    }
    
    public static class Routing {
        public List<RoutingRule> rules;
    }
    
    public static class RoutingRule {
        public String type;
        public String outboundTag;
        public String network;
    }
}

File: kt/app/src/main/kotlin/com/myvpn/simple/xray/XrayConfig.kt
package com.myvpn.simple.xray

data class XrayConfig(
    var inbounds: List<Inbound>? = null,
    var outbounds: List<Outbound>? = null,
    var log: Log = Log(),
    var routing: Routing? = null
) {
    data class Log(
        var loglevel: String = "warning"
    )

    data class Inbound(
        var protocol: String = "",
        var port: Int = 0,
        var listen: String = "",
        var settings: Map<String, Any>? = null,
        var sniffing: Sniffing? = null
    )

    data class Outbound(
        var tag: String = "",
        var protocol: String = "",
        var settings: Any? = null,
        var streamSettings: StreamSettings? = null
    )

    data class Sniffing(
        var enabled: Boolean = false,
        var destOverride: List<String>? = null,
        var routeOnly: Boolean = false
    )

    data class StreamSettings(
        var network: String = "",
        var security: String = "",
        var tlsSettings: TlsSettings? = null
    )

    data class TlsSettings(
        var serverName: String = "",
        var allowInsecure: Boolean = false
    )

    data class Routing(
        var rules: List<RoutingRule>? = null
    )

    data class RoutingRule(
        var type: String = "",
        var outboundTag: String = "",
        var network: String = ""
    )
}
